File: 020/020.cpp
// Find the sum of the digits in the number 100!

#include<iostream>
#include<string>
using namespace std;

double factorial(int n)
{
  if(n==0) return 1;
  return n*factorial(n-1);
}

int main()
{
  // Might have to use bigger ints for this.
  // Won't work with current implementation
  string s = to_string(factorial(100));
  cout << s << endl;
  int sum = 0;
  int i=0;
  while(s[i] != '.')
  {
    sum += (int)s[i] - '0';
    i++;
  }
  cout << sum << endl;
  return 0;
}

// Remember to store 100! in the biggest possible number type
// and in this case it's not unsigned long long int but double

File: 020/020.py
# -*- coding: utf-8 -*-

'''
n! means n × (n − 1) × ... × 3 × 2 × 1

For example, 10! = 10 × 9 × ... × 3 × 2 × 1 = 3628800,
and the sum of the digits in the number 10! is 3 + 6 + 2 + 8 + 8 + 0 + 0 = 27.

Find the sum of the digits in the number 100!
'''

def fac(n):
  if n==0:
    return 1
  return n*fac(n-1)

# Convert to string
s = str(fac(100))

# Put into list of characters
l = list(s)

sum = 0
for i in l:
  sum += int(i)

print sum
